File: server/app/model/schema/question-choice/question-choice.ts
import { SchemaFactory } from '@nestjs/mongoose';
import { Document } from 'mongoose';

export type QuestionChoiceDocument = QuestionChoice & Document;

export class QuestionChoice {
    text: string;
    isCorrect?: boolean;
    isSelected?: boolean;
}

// eslint-disable-next-line @typescript-eslint/naming-convention
export const QuestionChoiceSchema = SchemaFactory.createForClass(QuestionChoice); // Respecter la convention de nommer les classes en UpperCamelCase

File: client_leger/lib/models/question_choice.dart
class QuestionChoice {
  String text;
  bool? isCorrect;
  bool? isSelected;

  QuestionChoice({
    required this.text,
    this.isCorrect,
    this.isSelected,
  });

  factory QuestionChoice.fromJson(Map<String, dynamic> json) {
    return QuestionChoice(
      text: json['text'],
      isCorrect: json['isCorrect'],
      isSelected: json['isSelected'] ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'text': text,
      'isCorrect': isCorrect,
      'isSelected': isSelected,
    };
  }
}
